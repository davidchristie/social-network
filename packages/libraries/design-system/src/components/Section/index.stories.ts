import Section from ".";
import addComponentStory from "../../utilities/stories/addComponentStory";

addComponentStory(Section);
