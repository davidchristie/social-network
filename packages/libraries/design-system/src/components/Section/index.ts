import React from "react";

import elementWithClass from "../../utilities/elementWithClass";
import "./index.css";

interface Props extends React.HTMLAttributes<HTMLElement> { }

const Section = elementWithClass<Props>("Section", "section");

export default Section;
