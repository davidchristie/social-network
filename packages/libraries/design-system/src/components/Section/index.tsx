import React from "react";

import "./index.css";

interface Props extends React.HTMLAttributes<HTMLElement> { }

const Section: React.SFC<Props> = ({ className, ...attributes }) => (
  <section
    className={`Section${className ? " " + className : ""}`}
    {...attributes}
  />
);

export default Section;
