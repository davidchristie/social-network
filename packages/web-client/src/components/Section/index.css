.Section {
  border: solid black 2px;
  padding: 1em;
}

@media only screen and (min-width: 600px) {
  .Section {
    padding: 2em;
  }
}
