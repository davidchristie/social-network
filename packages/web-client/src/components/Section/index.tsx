import React from "react";

import "./index.css";

interface Props extends React.HTMLAttributes<HTMLElement> { }

export default class Section extends React.Component<Props> {
  public render () {
    const { children, className, ...attributes } = this.props;
    return (
      <section
        className={`Section${className ? " " + className : ""}`}
        {...attributes}
      >
        {children}
      </section>
    );
  }
}
