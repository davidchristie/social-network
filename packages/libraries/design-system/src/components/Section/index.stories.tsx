import { action } from "@storybook/addon-actions";
import { withInfo } from "@storybook/addon-info";
import { storiesOf } from "@storybook/react";
import React from "react";

import Section from ".";

const stories = storiesOf("Components", module);

stories.add(
  "Section",
  withInfo({ inline: true })(() => <Section
    children="Section"
    onClick={action("onClick")}
  />)
);
