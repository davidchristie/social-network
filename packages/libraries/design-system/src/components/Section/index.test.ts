import {
  describeWithCustomClass,
  describeWithNoProps
} from "test-utilities";

import Section from "./index";

describe("Section component", () => {
  describeWithNoProps(Section);

  describeWithCustomClass(Section);
});
