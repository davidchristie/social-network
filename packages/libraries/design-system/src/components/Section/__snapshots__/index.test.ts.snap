// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component with custom class matches snapshot 1`] = `
<section
  className="Section CUSTOM_CLASS"
/>
`;

exports[`Section component with no props matches snapshot 1`] = `
<section
  className="Section"
/>
`;
